src/gentropy/dataset/summary_statistics.py

```diff
@@ -103,3 +104,25 @@ def exclude_region(self: SummaryStatistics, region: str) -> SummaryStatistics:
             ),
             _schema=SummaryStatistics.get_schema(),
         )
+
+    def get_locus_sumstats(
+        self: SummaryStatistics,
+        locus: DataFrame,
```

Comment:
I don't really like this. Expecting a dataframe is a very vague requirement. It might won't have chromosome and position column in which case the function  fails. I believe it is more robust if we need studyid, chromosome, position. Yes, I think studyid should be mandated, otherwise the returned data might cause surprises.